### PR TITLE
Add complete documentation to the public interface

### LINKED ISSUE #5: Add inline documentation
The public interface in ListItemFormatter.swift needs fully documenting as a priority

The Internal and Private interface also needs documenting to some degree

### PR DESCRIPTION
**Pull request by liamnichols**
Closes #5

### CODE REVIEW

File: Source/ListItemFormatter.swift [hunk 1]
```diff
@@ -24,60 +24,116 @@ import Foundation
...
+        case or
+
+        /// A format mode suitable for arbitrary placeholder units with no mode (i.e
+        ///  measurement units).
+        case unit
     }
 
-    /// TBC
+    /// The following constants specify predefined format styles for list items.
     @objc(LNListItemFormatterStyle) public enum Style: Int {
```
Comment (bcylin) [inline Source/ListItemFormatter.swift:51]: [nits] Maybe it's worth mentioning `Mode` and `Style` are for constructing the keys like `standard`, `standardNarrow`, and `standardShort` used in the [plists](https://github.com/liamnichols/ListItemFormatter/blob/master/Resources/cldr_plists/en.plist)?

File: Source/ListItemFormatter.swift [hunk 2]
```diff
@@ -24,60 +24,116 @@ import Foundation
...
+    /// - Parameter list: The list items to be formatted into a localised, human
+    ///  readable list.
     ///
-    /// - Parameter list: TBC
-    /// - Returns: TBC
+    /// - Returns: A string representation of `list` formatted using the receiver’s
+    ///  current settings.
     @objc public func string(from list: [String]) -> String? {
         return string(for: list)
     }
```
Comment (bcylin) [inline Source/ListItemFormatter.swift:109]: [Q] Do you think it's a good idea to overload the method in Swift and expose a different name to Objective-C?

```swift
@objc(stringFrom:)
func string(for list: [String]) -> String?
```

**Review by bcylin — APPROVED** [review]
Looks nice and clear 👍

File: Source/ListItemFormatter.swift (hunk 1 above)
Comment (liamnichols) [reply Source/ListItemFormatter.swift:51]: My original intention was to not mention any specifics on the underlying implementation but thinking about it, I don't see no harm so I'll give it a go 👍

File: Source/ListItemFormatter.swift (hunk 2 above)
Comment (liamnichols) [reply Source/ListItemFormatter.swift:109]: Ahh I didn't think about this. I already opened #17 for improvements to the return value so can add this to that issue and tackle it separately to the documentation, thanks!

**Pull request merged**